### DISCUSSION TITLE
filament-tables: filter dropdown flashing in chrome

### DISCUSSION THREAD

**Discussion by jrmypttrsn** [filamentphp/filament#3913] [Help] [answered]
I've got a weird thing happening only in production and only in Chrome. 

I have a column that uses a view to display a button, which is using ```wire:poll```. While the polling occurs the select dropdowns for my table filters become unclickable.

![wire poll filter chrome bug](https://user-images.githubusercontent.com/16291625/189216768-84297872-18f4-4c16-840e-b638dd2d0633.gif)

Again, this doesn't occur locally, even in Chrome. I'm stumped on this one.

**Comment by studiowizjo** [comment]
Maybe try adding to it `wire:ignore`. If this won`t work, please provide some code to say something more.

**Reply by jrmypttrsn** [reply]
```wire:ignore``` does nothing as it's the select filter having the issue, not the button - there's no option to pass that to the filter. 

```php
protected function getTableFilters(): array
{
    return [
        SelectFilter::make('certifier_name')
            ->relationship('certifier', 'certifier_name')
            ->label('Certifier'),
        SelectFilter::make('operation_certification_status')
            ->label('Status')
            ->options([
                'Certified' =>'Certified',
                'Surrendered' =>'Surrendered',
                'Suspended' =>'Suspended',
                'Revoked' =>'Revoked'
            ]),
    ];
}

protected function getTableColumns(): array
{
    return [
        ViewColumn::make('operation_id')
            ->label('')
            ->alignCenter()
            ->view('filament.tables.columns.button'),
...
}
```

button.blade.php
``` html
<div :key="time().$getState()">
    <div wire:poll.1s>
        @if(condition)
            <div class="has-tooltip">
                <x-icon.badge-check class="text-green-500 has-tooltip"/>
                <span class="tooltip rounded-xl shadow-lg p-2 bg-green-100 text-sm text-green-700 border-green-700 font-bold -mt-10">
                This operation is on your watchlist
            </span>
            </div>
        @elseif(other_condition)
            <div class="has-tooltip">
                <x-icon.warning class="text-red-500 has-tooltip"/>
                <span class="tooltip rounded-xl shadow-lg p-2 bg-red-100 text-sm text-red-700 border-red-700 font-bold -mt-10">
                You have reached your limit of watched operations
            </span>
            </div>
        @else
            <div class="flex flex-col">
                <x-button.primary wire:click="add('{{ $getRecord()->getKey() }}')" wire:key="operation-{{ $getState() }}" class="has-tooltip p-2 mx-2">
                    <x-icon.plus/>
                    <span class="tooltip rounded-xl shadow-lg p-2 bg-gray-200 text-sm text-gray-700 -mt-10">
                        Add to Team Watchlist
                    </span>
                </x-button.primary>
            </div>
        @endif
    </div>
</div>
```

**Comment by danharrin** [comment]
Have you tried wrapping a non-Filament `<select>` in a wire:poll before? Is this a Livewire or a Filament bug?

**Reply by jrmypttrsn** [reply]
To clarify, it's the button that is using ```wire:poll```. That *seems* to be affecting the select dropdown generated by my filter but I can't say for certain as it's only happening in production (deployed through Forge on DigitalOcean) and only in Chrome browsers. 

This was only brought to my attention a couple of days ago by a user, so I'm not certain where the bug is originating. My guess is that it has something to do with Chrome and *maybe* a cache setting in DigitalOcean? But I thought I would ask here first to see if anyone has run into this particular issue, as edge as it is.

**Comment by jrmypttrsn** [comment, marked as answer]
I found a solution.

My LW component Blade file had two root elements. 🤦🏻‍♂️

Changed
```html
<div :key="time().$getState()">
    <div wire:poll.1s>
    {{-- BUTTON CODE --}}
    </div>
</div>
```
to
```html
<div :key="time().$getState()" wire:poll.250ms>
    {{-- BUTTON CODE --}}
</div>
```
and things work in Chrome again as expected.

Still not sure why this was only causing issues in Chrome and not other browsers but going to get on with my life.

**Reply by studiowizjo** [reply]
I guess the reason was using `time()` in `:key`. It should not change between requests, to avoid diffing issues.
